src/cmap/connect.ts

```diff
@@ -247,15 +247,6 @@ function parseSslOptions(options: ConnectionOptions): TLSConnectionOpts {
     }
   }
 
-  // Override checkServerIdentity behavior
-  if (!options.checkServerIdentity) {
-    // Skip the identity check by retuning undefined as per node documents
-    // https://nodejs.org/api/tls.html#tls_tls_connect_options_callback
-    result.checkServerIdentity = () => undefined;
-  } else if (typeof options.checkServerIdentity === 'function') {
-    result.checkServerIdentity = options.checkServerIdentity;
-  }
```

Comment:
`checkServerIdentity` is a nodejs TLS option, and nodejs asserts that it is of type function. I suggest that we don't intervene on a boolean type but allow the option to pass directly to the tls API.

Reply:
Seems fine for 4.0, but make sure the documentation reflects this, and that your upgrade notes mention it. The change can't be made for 3.x unfortunately.